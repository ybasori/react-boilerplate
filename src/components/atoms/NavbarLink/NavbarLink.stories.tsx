import type { Meta, StoryObj } from "@storybook/react";
import NavbarLink from "./NavbarLink";
import { MemoryRouter } from "react-router-dom";

//ðŸ‘‡ This default export determines where your story goes in the story list
const meta: Meta<typeof NavbarLink> = {
  title: "Atoms/NavbarLink",
  component: (props) => (
    <MemoryRouter>
      <NavbarLink {...props} />
    </MemoryRouter>
  ),
};

export default meta;
type Story = StoryObj<typeof NavbarLink>;

export const DefaultButton: Story = {
  args: {
    name: "button",
    to: "/",
  },
};
