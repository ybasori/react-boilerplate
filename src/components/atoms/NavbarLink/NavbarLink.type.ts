export interface IProps {
  to: string;
  name: string;
}
