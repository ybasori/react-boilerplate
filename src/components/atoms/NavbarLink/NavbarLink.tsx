import { Link } from "react-router-dom";
import { IProps } from "./NavbarLink.type";

const NavbarLink: React.FC<IProps> = ({ to, name }) => {
  return (
    <>
      <li>
        <Link to={to}>{name}</Link>
      </li>
    </>
  );
};

export default NavbarLink;
