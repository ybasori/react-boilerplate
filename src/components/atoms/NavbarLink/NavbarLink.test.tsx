import { fireEvent, render, screen } from "@testing-library/react";
import NavbarLink from "./NavbarLink";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

describe("Atoms/NavbarLink", () => {
  test("renders the button with the correct label", () => {
    const history = createMemoryHistory();
    render(
      <Router history={history}>
        <NavbarLink to="/" name="Click" />
      </Router>
    );

    // Assert that the button is in the document and has the correct text
    const buttonElement = screen.getByText("Click");
    expect(buttonElement).toBeInTheDocument();
  });

  test("calls the onClick handler when the button is clicked", () => {
    const history = createMemoryHistory();
    render(
      <Router history={history}>
        <NavbarLink to="/test" name="Click" />
      </Router>
    );

    const buttonElement = screen.getByText("Click");
    fireEvent.click(buttonElement);
    expect(history.location.pathname).toBe("/test");
  });

  test("Snapshot", () => {
    const history = createMemoryHistory();
    const { container } = render(
      <Router history={history}>
        <NavbarLink to="/" name="Click" />
      </Router>
    );

    expect(container).toMatchSnapshot();
  });
});
