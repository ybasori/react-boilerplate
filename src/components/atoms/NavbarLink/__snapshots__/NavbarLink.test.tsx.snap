// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/NavbarLink Snapshot 1`] = `
<div>
  <li>
    <a
      href="/"
    >
      Click
    </a>
  </li>
</div>
`;
